src/unix/unix_c/unix_get_pw_gr_nam_id_job.c

```diff
@@ -84,6 +85,15 @@ static value alloc_group_entry(struct group *entry)
         job->buffer = (char *)lwt_unix_malloc(buffer_size);            \
         job->result = FUNC##_r(job->ARG, &job->entry, job->buffer,     \
                                buffer_size, &job->ptr);                \
+        while (job->result == -1 && errno == ERANGE)                   \
+        {                                                              \
+            free(job->buffer);                                         \
+            buffer_size = buffer_size * 2;                             \
+            if (buffer_size < 0) break;                                \
```

Comment:
This condition is always false, because `buffer_size` is a `size_t`, which is unsigned, and so is never less than zero. I suppose this should be a bound check against some large buffer size instead.

Reply:
I'm checking if it's bigger than 2^30. I picked that constant sort of randomly: I wanted something less than 2^32 for compatibility with 32-bit architectures, but otherwise not real reason.